app/src/main/java/de/danoeh/antennapod/fragment/FeedItemlistFragment.java

```diff
@@ -562,7 +562,7 @@ private void loadItems() {
         if (feed != null && feed.getItemFilter() != null) {
             DBReader.loadAdditionalFeedItemListData(feed.getItems());
             FeedItemFilter filter = feed.getItemFilter();
-            feed.setItems(filter.filter(feed.getItems()));
+            feed.setItems(FeedItemUtil.filter(feed.getItems(), filter));
```

Comment:
Is this the only place where the filter is used? If that's the case, I would rather use the new SQL-based filter method (see #4563) because that reduces code duplication and makes filtering more efficient (as it's left to the database instead of retrieving everything and then throwing away items).

Reply:
I agree that the database query could be optimized. After I had a look in the code this doesn't seem to be a trivial change and needs some unit tests. Can we keep that separated from this pull request?

Reply:
When removing the Java-based filter code later anyway, I don't think we need to move it to another class first, to be honest.

Reply:
I implemented SQL-based filtering in #4985. Closing this PR now. The effect is the same (removed dependency) but it's faster and has less code duplication.